From 6d06d5ae80036f77149cc0a054ca13ec8c501271 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Wed, 25 Oct 2017 04:23:42 +0300
Subject: [PATCH 2048/2241] crypto: inside-secure: EBUSY is not an error on
 async request

When calling crypto_ahash_update it is OK to return EBUSY since
there may not be available resource but the request is in queue,
code should not treat EBUSY as an error.

Change-Id: Id231d45d9e4c4fffcdc858738698bdded1caa6a7
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45570
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/crypto/inside-secure/hash.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/inside-secure/hash.c b/drivers/crypto/inside-secure/hash.c
index 8cc7358..da0db44 100644
--- a/drivers/crypto/inside-secure/hash.c
+++ b/drivers/crypto/inside-secure/hash.c
@@ -821,7 +821,7 @@ static int safexcel_hmac_prepare_pad(struct ahash_request *areq, u8 *pad,
 	req->last_req = 1;
 	req->finish = finish;
 	ret = safexcel_ahash_update(areq);
-	if (ret && ret != -EINPROGRESS)
+	if (ret && ret != -EINPROGRESS && ret != -EBUSY)
 		return ret;
 
 	wait_for_completion_interruptible(&result.completion);
-- 
2.7.4

